### PR TITLE
Pix 124 p3

### PR DESCRIPTION
**Pull request by pirminb97**

### CODE REVIEW

File: lib/src/router.dart [hunk 1]
```diff
@@ -71,6 +90,24 @@ class AppRouter {
...
+    Matrix4 matrix = Matrix4.identity();
+    if (args != null &&
+        (args is Map) &&
+        args.containsKey(transformationMatrixArg)) {
+      matrix = args[transformationMatrixArg];
+    }
+    return matrix;
+  }
+
+  ImageFilterResult _getImageFromArgs(dynamic args) {
```
Comment (LeonidArefev) [inline lib/src/router.dart:103]: no please, use singlotone here, dont pass image as parameter.

File: lib/src/router.dart [hunk 2]
```diff
@@ -71,6 +90,24 @@ class AppRouter {
     return path;
   }
 
+  Matrix4 _getDetailsFromArgs(dynamic args) {
+    Matrix4 matrix = Matrix4.identity();
+    if (args != null &&
+        (args is Map) &&
+        args.containsKey(transformationMatrixArg)) {
```
Comment (LeonidArefev) [inline lib/src/router.dart:97]: about transfering matrix as parameter we also need to tolk with Thomas. Idea to use complex objects as arguments have some ideology problems

File: lib/src/screens/image/widgets/custom_shape.dart [hunk 3]
```diff
@@ -33,7 +33,7 @@ class ShapePainter extends CustomPainter {
 
   void _drawImageSelection(Canvas canvas, Size size) {
     var paint1 = Paint()
-      ..color = AppTheme.primaryColor
+      ..color = isImageSelected ? Colors.transparent : AppTheme.primaryColor
```
Comment (LeonidArefev) [inline lib/src/screens/image/widgets/custom_shape.dart:36]: queston, do we have situations, when selectedPosition is valid number and image selected at the same time?
may be we dont need isImageSelected as parameter to class, and you can calculate it

imageSelected=selectedPosition <0||selectedPosition >=positions.length

File: lib/src/screens/image/widgets/image_viewer.dart [hunk 4]
```diff
@@ -52,6 +54,7 @@ class ImageViewer extends StatelessWidget {
       onLongPressMoveUpdate: onMoveFilter,
       onLongPressStart: onLongPressStart,
       child: InteractiveViewer(
+          onInteractionUpdate: this.onInteractionUpdate,
```
Comment (LeonidArefev) [inline lib/src/screens/image/widgets/image_viewer.dart:57]: if we dont use it for creation current transformatoin state, may be we dont need this event-handler?

**Review by LeonidArefev — CHANGES_REQUESTED** [review]

File: lib/src/router.dart (hunk 1 above)
Comment (tkuenneth) [reply lib/src/router.dart:103]: Leonid, please do not just request changes. Explain why you think what Pirmin does is not ok. Without further information I question the use of a singleton here

Comment (LeonidArefev) [reply lib/src/router.dart:103]: i suppose that tranfering complext parameters in navigation its not very good idea. in next comment below a wrote, that we need to ask you about that.
Both comments are about same situation.

``about transfering matrix as parameter we also need to tolk with Thomas. Idea to use complex objects as arguments have some ideology problems``

Comment (tkuenneth) [reply lib/src/router.dart:103]: But then a singleton is not a good idea either. We must refraim from pre-optimization. The best approach is always to start with the parameters that seem natural. So Pirmins approach was quite alright. We change things only if we run into problems.

File: lib/src/screens/image/widgets/image_viewer.dart (hunk 4 above)
Comment (pirminb97) [reply lib/src/screens/image/widgets/image_viewer.dart:57]: it prepared for later implementation. this might be removed later on

**Pull request merged**